src/FileStorageBase/Invoker/ContractInvoker.cs

```diff
@@ -0,0 +1,133 @@
+using System;
+using System.Linq;
+using Akka.Actor;
+using Neo.Cryptography.ECC;
+using Neo.IO;
+using Neo.Ledger;
+using Neo.Network.P2P.Payloads;
+using Neo.Persistence;
+using Neo.SmartContract;
+using Neo.SmartContract.Native;
+using Neo.VM;
+using Neo.Wallets;
+
+namespace Neo.FileStorage.Invoker
+{
+    public abstract class ContractInvoker
+    {
+        public Wallet Wallet { get; init; }
+        public NeoSystem NeoSystem { get; init; }
+        public IActorRef Blockchain { get; init; }
+
+        protected void Invoke(UInt160 contractHash, string method, long fee, params object[] args)
+        {
+            InvokeResult result = TestInvoke(contractHash, method, args);
+            using SnapshotCache snapshot = NeoSystem.GetSnapshot();
+            uint height = NativeContract.Ledger.CurrentIndex(snapshot);
+            Random rand = new();
+            Transaction tx = new()
+            {
+                Version = 0,
+                Nonce = (uint)rand.Next(),
+                Script = result.Script,
+                ValidUntilBlock = height + NeoSystem.Settings.MaxValidUntilBlockIncrement,
+                Signers = new Signer[] { new Signer() { Account = Wallet.GetAccounts().ToArray()[0].ScriptHash, Scopes = WitnessScope.Global } },
```

Comment:
could not have gas and in the log we don't show the sender

Reply:
If there is no gas `TestInvoke` will throw exception;
https://github.com/neo-ngd/neo-plugins/blob/f0a20a15298afd6322c97a4b16f50b6e709807cb/src/FileStorageBase/Invoker/ContractInvoker.cs#L24